pkg/registry/consul.go

```diff
@@ -216,15 +218,23 @@ func (c *ConsulClient) Put(ctx context.Context, key string, value []byte) error
 		registryOperationTimeTaken.WithLabelValues(env, "Put").Observe(time.Now().Sub(startTime).Seconds())
 	}()
 
-	_, err := c.client.KV().Put(&api.KVPair{
-		Key:   key,
-		Value: value,
-	}, nil)
-	return err
+	// Using transaction because a simple put does not fetch the modifyindex
+	putTxn := api.TxnOps{&api.TxnOp{KV: &api.KVTxnOp{Verb: api.KVSet, Key: key, Value: value}}}
+	ok, resp, _, err := c.client.Txn().Txn(putTxn, nil)
```

Comment:
check `err!=nil {}` as well

Reply:
It's being done inside `if !ok`
If err is not nil, then ok will not be true

Reply:
// If there is a problem making the transaction request then an error will be
// returned. Otherwise, the ok value will be true if the transaction succeeded
// or false if it was rolled back.
Was going through API doc, seems correct 👍🏼